static/src/javascripts/projects/common/modules/analytics/css-logging.js

```diff
@@ -64,15 +66,27 @@ define([
         }), allRules);
     }
 
-    function sendReports(sendAll) {
-        setTimeout(function () {
-            _.each(sendAll ? getStylesheets() : [randomStylesheet()], function (stylesheet) {
-                sendReport(stylesheet, sendAll);
-            });
-        }, 3000);
+    function makeSender(sendAll) {
+        return _.debounce(function (clickSpec) {
+            if (!clickSpec || clickSpec.samePage) {
+                setTimeout(function () {
+                    _.each(sendAll ? getStylesheets() : [getRandomStylesheet()], function (stylesheet) {
+                        sendReport(stylesheet, sendAll);
+                    });
+                }, _.random(0, 3000));
+            }
+        }, 300);
     }
 
-    return {
-        run: sendReports
+    return function () {
+        var sendAll = window.location.hash === '#csslogging',
+            sender;
+
+        if (sendAll || _.random(1, 5000) === 1) {
+            sender = makeSender(sendAll);
+            sender();
```

Comment:
I need `sender` as an unexecuted function, to attach to the event listeners in the following two lines.